src/foam/doc/DocBrowser.js

```diff
@@ -532,3 +559,954 @@ foam.debug.doc = function(opt_obj, showUnused) {
+          width: w || 200,
+          height: h || 30,
+          color: '#ffffe0', //this.UNSELECTED_COLOR
+          border: 'black'
+        });
+
+        this.setData(extendsBox.x, extendsBox.y, cls.id);
+
+        var extendsNameLabel = foam.graphics.Label.create({
+          align: 'center',
+          x: x,
+          y: y - ((i + 1) * d) - marge,
+          color: 'black',
+          font: '20px Arial',
+          width: w || 200,
+          height: h || 30,
+          text: cls.name
+        });
+
+        extendsLine = foam.graphics.Line.create({
+          startX: x + extendsBox.width / 2 || 0,
+          startY: y - (d * i),
+          endX: x + extendsBox.width / 2 || 0,
+          endY: y - (d * i) - (extendsBox.height * 2) + this.triangleSize,
+          color: 'black',
+          lineWidth: 2
+        });
+
+        var triangleEndExtends = this.triangle(x + extendsBox.width / 2, y - (d * i) - (extendsBox.height * 2),0);
+
+        this.selected = this.canvas.addChildren(extendsBox, extendsNameLabel, extendsLine, triangleEndExtends);
+
+        this.setData(extendsBox.x, extendsBox.y, cls);
+
+        if (cls === foam.core.FObject) break;
+      }
+    },
+
+    function addImplements(x, y, w, h) {
+      var marge = 5;
+      var sideY = 150; //d
+      var sideX = -250;
+      var cls = this.data;
+
+      if (cls.model_.implements !== undefined) {
+        for (var key in cls.model_.implements) {
+          var a = cls.model_.implements[key];
+          if (a.path !== undefined) {
+            var implementsName = this.Box.create({
+              x: x - sideX,
+              y: y - sideY,
+              width: w || 200,
+              height: h || 30,
+              color: '#ffffe0', //this.UNSELECTED_COLOR
+              border: 'black'
+            });
+
+            this.setData(implementsName.x, implementsName.y, a.path);
+
+            var implementsNameLabel = foam.graphics.Label.create({
+              align: 'center',
+              x: x - sideX + this.dashedstep,
+              y: y - sideY - marge,
+              color: 'black',
+              font: '20px Arial',
+              width: w || 200,
+              height: h || 30,
+              text: a.path
+            });
+
+            var dashedLine = this.dashedLine(sideY, x + implementsName.width * 3 / 4 + this.dashedstep, y - this.dashedstep, x + sideY + (implementsName.width * 3 / 4) - implementsName.height, y - sideY + implementsName.height);
+
+            var triangleEndImplement = this.triangle(x + sideY + implementsName.width * 3 / 4 - implementsName.height, y - sideY + implementsName.height,0);
+
+            this.selected = this.canvas.addChildren(implementsName, implementsNameLabel, triangleEndImplement);
+          }
+        }
+      }
+    },
+
+    function addRequires(x, y, w, h) {
+      var triangleSize = 5;
+      var marge = 5;
+      var d = 300;
+      var cls = this.data;
+      if (cls.model_.requires !== undefined) {
```

Comment:
Here and elsewhere, see the FOAM style guideline regarding spacing: https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md